### PR TITLE
[core] Increased FormatTime precision

### PR DESCRIPTION
**Pull request by maxsharabayko**
The internal clock might have a nanosecond precision or any other below 1 microsecond.
The FormatTime function used for logging prints only a microsecond precision, which might be not enough for some scenarios.

This PR improves the precision of the `FormatTime` function to the actual precision of the clock used.

Example output: `05:53:45.21225876683803 [STDY]`.

### Other changes

- Internal constant `s_cpu_frequency` (sync_posix.cpp) renamed to `s_clock_ticks_per_us`.

### TODO

- [x] Find a better name for `clock_subsecond_precision()` function. Found: `clockSubsecondPrecision()`.
- [x] Fix sync FormatTime unit test

### CODE REVIEW

File: srtcore/sync_cxx11.cpp [hunk 1]
```diff
@@ -16,6 +16,34 @@
 #include "srt_compat.h"
 #include "common.h"
 
+////////////////////////////////////////////////////////////////////////////////
```
Comment (ethouris) [inline srtcore/sync_cxx11.cpp:19]: Put this into an anonymous namespace to avoid applicationwise name clash risk.

File: srtcore/sync_posix.cpp [hunk 2]
```diff
@@ -114,8 +114,18 @@ int64_t get_cpu_frequency()
     return frequency;
 }
 
-const int64_t s_cpu_frequency = get_cpu_frequency();
+int caclulate_subsecond_signs(int64_t ticks_per_us)
+{
+    int signs = 0;
+    while (ticks_per_us /= 10) ++signs;
```
Comment (ethouris) [inline srtcore/sync_posix.cpp:120]: I think you could speed this up by using "binary search" method and start from the least value that you expect to have in all implementations (I guess it's 6). And the better name for it is "precission".

File: srtcore/sync_cxx11.cpp (hunk 1 above)
Comment (ethouris) [inline srtcore/sync_cxx11.cpp:19]: Better put this into an anonymous namespace to avoid name clash risk.

File: srtcore/sync_posix.cpp (hunk 2 above)
Comment (maxsharabayko) [reply srtcore/sync_posix.cpp:120]: This function is called only once, and the value is then stored in `s_clock_subsecond_signs`.
No need for optimization.

File: srtcore/sync_cxx11.cpp (hunk 1 above)
Comment (maxsharabayko) [reply srtcore/sync_cxx11.cpp:19]: Done.

File: srtcore/sync_posix.cpp (hunk 2 above)
Comment (ethouris) [reply srtcore/sync_posix.cpp:120]: Ah, there's no problem then.

Still, I think `calculate_subsecond_prec` should be better and `clock_subsecond_precision()` is a good enough name, I'd only be picky about the naming convention (`clockSubsecondPrecision()`).

Comment (maxsharabayko) [reply srtcore/sync_posix.cpp:120]: I have already renamed `aclulate_subsecond_signs(..)`  to `count_subsecond_precision(..)`. Not sure why GitHub still shows the old naming here.

Comment (maxsharabayko) [reply srtcore/sync_posix.cpp:120]: Renamed `clock_subsecond_precision()` to `clockSubsecondPrecision()`

**Review by ethouris — APPROVED** [review]

**Pull request merged**